### DISCUSSION TITLE
Package-diff and Changelogs for gitlab packages registry

### DISCUSSION THREAD

**Discussion by robin-ln** [renovatebot/renovate#29289] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

gitlab

### Please tell us more about your question or problem

[my repo](https://gitlab.com/robin.louarn/monorepo-test)

 my package [@my-company/plugin-hello](https://gitlab.com/robin.louarn/monorepo-test/-/blob/main/packages/plugin-hello/package.json?ref_type=heads)

with this config in [package.json](https://gitlab.com/robin.louarn/monorepo-test/-/blob/main/packages/plugin-hello/package.json?ref_type=heads) : 

```json
  "repository": {
    "type": "git",
    "url": "https://gitlab.com/robin.louarn/monorepo-test.git",
    "directory": "packages/plugin-hello"
  },
```

but when i run this command : 
```
GITHUB_COM_TOKEN=$GITHUB_COM_TOKEN LOG_LEVEL=debug pnpm renovate \
    --platform=gitlab \
    --token $GITLAB_TOKEN \
    --endpoint $CI_API_V4_URL \
    --npmrc "@my-company:registry=https://gitlab.com/api/v4/projects/56880161/packages/npm/\n//gitlab.com/api/v4/projects/56880161/packages/npm/:_authToken=$CI_JOB_TOKEN" \
    $CI_PROJECT_PATH
```

[my logs](https://github.com/renovatebot/renovate/files/15462221/tmp.txt)

this [MR](https://gitlab.com/robin.louarn/monorepo-test/-/merge_requests/55) was generated

but i have no [changelogs](https://gitlab.com/robin.louarn/monorepo-test/-/blob/main/packages/plugin-hello/CHANGELOG.md?ref_type=heads) and infinite load on [package-diff](https://app.renovatebot.com/package-diff?name=@my-company%2fconfig-eslint&from=4.0.3&to=4.0.4)

can you help me ? 🙏

**Comment by rarkins** [comment, marked as answer]
The package-diff only works for npmjs.org packages.

Regarding changelogs, see https://docs.renovatebot.com/key-concepts/changelogs/

**Reply by robin-ln** [reply]
Sorry, could you guide me a little more? 

I have read this doc several times (and several discussion threads), but I cannot retrieve the changes logs.

It may be a matter of "sourUrl" or "metadata" but I don't know how to do this verification...

**Reply by robin-ln** [reply]
Hum ok, on GET http call on https://gitlab.com/api/v4/projects/56880161/packages/npm/@my-company/plugin-hello i have no repository metada, but i don't know why

**Reply by rarkins** [reply]
I think GitLab drops it in their response

**Reply by robin-ln** [reply]
yes i think its gitlab issue ... https://docs.gitlab.com/ee/user/packages/npm_registry/#npm-metadata
i will search more

**Reply by robin-ln** [reply]
https://gitlab.com/gitlab-org/gitlab/-/issues/344107
Ok it's maybe for the 17.3 release of gitlab 🥲
